Encumbrance Action Modifier & Current Mag Size Fix

Andre-G99 opened the pull request:
-Fixed the current mag size of a firearm going below 0
-Fixed encumbrance modifiers not being added to action rolls such as initiative, dodging, etc.

Commit:
encumbrance modifier fix

removed function addEncumbranceModifiers() implemented new solution in actor-degenesis.js

Commit:
Encumbrance Modifier Fix

added function prepareEncumbranceModifiers() in order to have encumbrance affect dice rolls for actions such as initiative, dodging, etc.

Commit:
Current Mag Size Fix

notifies player that mag is empty and returns null to prevent current mag size from going below 0

moo-man commented on module/actor/actor-degenesis.js:
```diff
@@ -303,10 +323,18 @@ export class DegenesisActor extends Actor {
+        }
+        // notifies player that mag is empty and returns null to prevent mag size from going below 0
+        else {
+            window.alert("Mag is empty);
```
Let's use the notification framework provided by Foundry, `window.alert` is a bit primitive

Andre-G99 commented:
Gotcha, I'll make the change later today.

On Sat, Jun 18, 2022, 11:51 AM Russell ***@***.***> wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In module/actor/actor-degenesis.js
> <https://github.com/moo-man/DEGENESIS-FoundryVTT/pull/52#discussion_r901005574>
> :
>
> >              this.updateEmbeddedDocuments("Item", [{ _id: rollData.weapon.id, "data.mag.current": rollData.weapon.mag.current - 1 }])
> -        this.postRollChecks(rollResults, "weapon")
> -        return { rollResults, cardData }
> +            this.postRollChecks(rollResults, "weapon")
> +            return { rollResults, cardData }
> +        }
> +        // notifies player that mag is empty and returns null to prevent mag size from going below 0
> +        else {
> +            window.alert("Mag is empty);
>
> Let's use the notification framework provided by Foundry, window.alert is
> a bit primitive
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/moo-man/DEGENESIS-FoundryVTT/pull/52#pullrequestreview-1011443488>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AFHSKMEAA3ZR7KI7JA7HDVDVPYLC5ANCNFSM5ZFC5QTQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

Commit:
Replaced UI For Warning PopUp

replaced original window.alert() code with foundry's ui.notifications.warn()

Andre-G99 commented:
Just pushed the commit for replacing window.alert with foundry notifications

Commit:
Added Notif For Encumbrance

Added a notification for when players over encumbered and continue to add weight

moo-man commented:
Why close this?
